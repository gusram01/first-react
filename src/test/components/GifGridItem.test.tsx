import React from 'react';
import GifGridItem from '../../components/GiFGridItem';
import { shallow, ShallowWrapper } from 'enzyme';

describe('GifGridItem tests', () => {
  const dummyTest = {
    id: 'idTest',
    title: 'Dummy Test',
    url: 'https://This-i-a-dummy-url'
  };
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  beforeEach(() => { wrapper = shallow(<GifGridItem img={dummyTest} />) });


  test('should render GifGridItem correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('paragraph should have title ', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(dummyTest.title);
  });

  test('image should have src & alt', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(dummyTest.url);
    expect(img.prop('alt')).toBe(dummyTest.title);
  });

  test('should have classname animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect(div.prop('className')?.includes('animate__fadeIn')).toBe(true);
  });
});