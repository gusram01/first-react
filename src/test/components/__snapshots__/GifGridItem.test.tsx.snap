// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests should render GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Dummy Test"
    src="https://This-i-a-dummy-url"
  />
  <p>
    Dummy Test
  </p>
</div>
`;
